// langchain@1.0.1/dist/index.js
import { __export } from "./_virtual/rolldown_runtime.js";
import { initChatModel } from "./chat_models/universal.js";
import { MultipleStructuredOutputsError, MultipleToolsBoundError, StructuredOutputParsingError, ToolInvocationError } from "./agents/errors.js";
import { ProviderStrategy, ToolStrategy, providerStrategy, toolStrategy } from "./agents/responses.js";
import { countTokensApproximately } from "./agents/middleware/utils.js";
import { createMiddleware } from "./agents/middleware.js";
import { FakeToolCallingModel } from "./agents/tests/utils.js";
import { createAgent } from "./agents/index.js";
import { summarizationMiddleware } from "./agents/middleware/summarization.js";
import { humanInTheLoopMiddleware } from "./agents/middleware/hitl.js";
import { anthropicPromptCachingMiddleware } from "./agents/middleware/promptCaching.js";
import { dynamicSystemPromptMiddleware } from "./agents/middleware/dynamicSystemPrompt.js";
import { llmToolSelectorMiddleware } from "./agents/middleware/llmToolSelector.js";
import { piiRedactionMiddleware } from "./agents/middleware/piiRedaction.js";
import { ClearToolUsesEdit, contextEditingMiddleware } from "./agents/middleware/contextEditing.js";
import { ToolCallLimitExceededError, toolCallLimitMiddleware } from "./agents/middleware/toolCallLimit.js";
import { TODO_LIST_MIDDLEWARE_SYSTEM_PROMPT, todoListMiddleware } from "./agents/middleware/todoListMiddleware.js";
import { modelCallLimitMiddleware } from "./agents/middleware/callLimit.js";
import { modelFallbackMiddleware } from "./agents/middleware/modelFallback.js";
import "./agents/middleware/index.js";
import { AIMessage, AIMessageChunk, BaseMessage, BaseMessageChunk, HumanMessage, HumanMessageChunk, SystemMessage, SystemMessageChunk, ToolMessage, ToolMessageChunk, filterMessages, trimMessages } from "@langchain/core/messages";
import { DynamicStructuredTool, DynamicTool, StructuredTool, Tool, tool } from "@langchain/core/tools";
import { InMemoryStore } from "@langchain/core/stores";
import { getContextVariable, setContextVariable } from "@langchain/core/context";
import { Document } from "@langchain/core/documents";

//#region src/index.ts
var src_exports = {};
__export(src_exports, {
	AIMessage: () => AIMessage,
	AIMessageChunk: () => AIMessageChunk,
	BaseMessage: () => BaseMessage,
	BaseMessageChunk: () => BaseMessageChunk,
	ClearToolUsesEdit: () => ClearToolUsesEdit,
	Document: () => Document,
	DynamicStructuredTool: () => DynamicStructuredTool,
	DynamicTool: () => DynamicTool,
	FakeToolCallingModel: () => FakeToolCallingModel,
	HumanMessage: () => HumanMessage,
	HumanMessageChunk: () => HumanMessageChunk,
	InMemoryStore: () => InMemoryStore,
	MultipleStructuredOutputsError: () => MultipleStructuredOutputsError,
	MultipleToolsBoundError: () => MultipleToolsBoundError,
	ProviderStrategy: () => ProviderStrategy,
	StructuredOutputParsingError: () => StructuredOutputParsingError,
	StructuredTool: () => StructuredTool,
	SystemMessage: () => SystemMessage,
	SystemMessageChunk: () => SystemMessageChunk,
	TODO_LIST_MIDDLEWARE_SYSTEM_PROMPT: () => TODO_LIST_MIDDLEWARE_SYSTEM_PROMPT,
	Tool: () => Tool,
	ToolCallLimitExceededError: () => ToolCallLimitExceededError,
	ToolInvocationError: () => ToolInvocationError,
	ToolMessage: () => ToolMessage,
	ToolMessageChunk: () => ToolMessageChunk,
	ToolStrategy: () => ToolStrategy,
	anthropicPromptCachingMiddleware: () => anthropicPromptCachingMiddleware,
	contextEditingMiddleware: () => contextEditingMiddleware,
	countTokensApproximately: () => countTokensApproximately,
	createAgent: () => createAgent,
	createMiddleware: () => createMiddleware,
	dynamicSystemPromptMiddleware: () => dynamicSystemPromptMiddleware,
	filterMessages: () => filterMessages,
	getContextVariable: () => getContextVariable,
	humanInTheLoopMiddleware: () => humanInTheLoopMiddleware,
	initChatModel: () => initChatModel,
	llmToolSelectorMiddleware: () => llmToolSelectorMiddleware,
	modelCallLimitMiddleware: () => modelCallLimitMiddleware,
	modelFallbackMiddleware: () => modelFallbackMiddleware,
	piiRedactionMiddleware: () => piiRedactionMiddleware,
	providerStrategy: () => providerStrategy,
	setContextVariable: () => setContextVariable,
	summarizationMiddleware: () => summarizationMiddleware,
	todoListMiddleware: () => todoListMiddleware,
	tool: () => tool,
	toolCallLimitMiddleware: () => toolCallLimitMiddleware,
	toolStrategy: () => toolStrategy,
	trimMessages: () => trimMessages
});

//#endregion
export { AIMessage, AIMessageChunk, BaseMessage, BaseMessageChunk, ClearToolUsesEdit, Document, DynamicStructuredTool, DynamicTool, FakeToolCallingModel, HumanMessage, HumanMessageChunk, InMemoryStore, MultipleStructuredOutputsError, MultipleToolsBoundError, ProviderStrategy, StructuredOutputParsingError, StructuredTool, SystemMessage, SystemMessageChunk, TODO_LIST_MIDDLEWARE_SYSTEM_PROMPT, Tool, ToolCallLimitExceededError, ToolInvocationError, ToolMessage, ToolMessageChunk, ToolStrategy, anthropicPromptCachingMiddleware, contextEditingMiddleware, countTokensApproximately, createAgent, createMiddleware, dynamicSystemPromptMiddleware, filterMessages, getContextVariable, humanInTheLoopMiddleware, initChatModel, llmToolSelectorMiddleware, modelCallLimitMiddleware, modelFallbackMiddleware, piiRedactionMiddleware, providerStrategy, setContextVariable, src_exports, summarizationMiddleware, todoListMiddleware, tool, toolCallLimitMiddleware, toolStrategy, trimMessages };
